fetch.js

```diff
@@ -105,9 +105,34 @@
     this.bodyUsed = false
 
     if (blobSupport) {
+      this._initBody = function(body) {
+        this._bodyInit = body
+        if (typeof body === 'string') {
+          this._bodyText = body
+        } else if (Blob.prototype.isPrototypeOf(body)) {
+          this._bodyBlob = body
+        } else if (FormData.prototype.isPrototypeOf(body)) {
+          this._bodyFormData = body
+        } else if (!body) {
+          this._bodyText = ''
+        } else {
+          throw new Error('unsupported BodyInit type')
```

Comment:
@annevk when handling `BodyInit` types, whats the correct way to handle unknown types? The spec just says "switch on object's type" with no default case.

Chrome Canary seems to just cast everything via `toString()`.

``` js
new Response(function() {}).text()
// "function () {}"
```